### PR TITLE
Add new ``--core-conf`` command line argument to allow passing `core.` confs via CLI

### LINKED ISSUE #13410: Add a way to define global confs from CLI
I think we will definitely need some command line ``-global-conf`` argument to be able to define core values like this one

_Originally posted by @memsharded in https://github.com/conan-io/conan/pull/13321#discussion_r1130707209_

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Feature: Add new ``--core-conf`` command line argument to allow passing `core.` confs via CLI.
Docs: https://github.com/conan-io/docs/pull/3515

Close https://github.com/conan-io/conan/issues/13410

### CODE REVIEW

File: conan/cli/command.py [hunk 1]
```diff
@@ -100,9 +102,14 @@ class ConanArgumentParser(argparse.ArgumentParser):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
```
Comment (AbrilRBS) [inline conan/cli/command.py:105]: What if we add the -gc arg in this file so that every command gets it by default? Just like for format

**Review by AbrilRBS — COMMENTED** [review]
Looking good! A suggestion I have, but yes let's talk about it!

File: conan/cli/command.py (hunk 1 above)
Comment (memsharded) [reply conan/cli/command.py:105]: It is being added to all commands already, isn't it?

Comment (memsharded) [reply conan/cli/command.py:105]: If you mean the parsing and handling, we need the ``conan_api``, which doesn't seem easily available here.

Comment (AbrilRBS) [reply conan/cli/command.py:105]: Oh missread which file was being modified, we actually had the same idea then, 👍

**Comment by AbrilRBS** [comment]
> I still have some concerns, most of the confs are really not for command line, but some it makes sense.

Yeah, I think we should warn for things like `core.cache:storage_path` which are not set early enough, right?

**Comment by memsharded** [comment]
> Yeah, I think we should warn for things like core.cache:storage_path which are not set early enough, right?

Actually, it is possible, I have extended the test to prove it

**Comment by AbrilRBS** [comment]
Ohhhhh, it actually is! This is great!

**Comment by memsharded** [comment]
I have changed this PR, no need to modify every ``parse_args`` call to inject the api, with this change, every single Conan command gets a ``-gc/--global-conf`` argument that injects all the conf the very first thing in the ``conan_api.config.global_conf`` that everybody else reads

File: conan/cli/command.py [hunk 2]
```diff
@@ -97,12 +99,18 @@ def _format(self, parser, info, *args):
...
 
     def parse_args(self, args=None, namespace=None):
         args = super().parse_args(args)
         ConanOutput.define_log_level(os.getenv("CONAN_LOG_LEVEL", args.v))
+        if args.global_conf:
+            from conans.model.conf import ConfDefinition
+            confs = ConfDefinition()
+            confs.loads("\n".join(args.global_conf))
+            self._conan_api.config.global_conf.update_conf_definition(confs)
```
Comment (AbrilRBS) [inline conan/cli/command.py:118]: This is great ⭐

**Review by AbrilRBS — APPROVED** [review]
😍

**Review by franramirez688 — APPROVED** [review]
This is great 👏

**Review by czoido — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Review by AbrilRBS — APPROVED** [review]

**Pull request merged**